pandas/core/arrays/period.py

```diff
@@ -141,21 +142,27 @@ def _simple_new(cls, values, freq=None, **kwargs):
         Ordinals in an ndarray are fastpath-ed to `_from_ordinals`
         """
 
+        if is_period_dtype(values):
+            freq = dtl.validate_dtype_freq(values.dtype, freq)
+            values = values.asi8
+
         if not is_integer_dtype(values):
             values = np.array(values, copy=False)
             if len(values) > 0 and is_float_dtype(values):
                 raise TypeError("{cls} can't take floats"
                                 .format(cls=cls.__name__))
-            return cls(values, freq=freq)
+            return cls(values, freq=freq, **kwargs)
 
-        return cls._from_ordinals(values, freq)
+        return cls._from_ordinals(values, freq=freq, **kwargs)
 
     @classmethod
-    def _from_ordinals(cls, values, freq=None):
+    def _from_ordinals(cls, values, freq=None, **kwargs):
         """
         Values should be int ordinals
         `__new__` & `_simple_new` cooerce to ordinals and call this method
         """
+        # **kwargs are included so that the signature matches PeriodIndex,
+        #  letting us share _simple_new
```

Comment:
Why do you want to share this? Once we actually don't inherit it anymore, we will need to add it back to PeriodIndex

Reply:
Eventually, yes, I think we can share a `_simple_new`. Regardless, I think the signatures for all these should match across index (and array) classes.

Reply:
But this one is about `_from_ordinals`, which will not be shared across index classes as it is Period-specific

Reply:
I think we're moving towards getting rid of `_from_ordinals` in order to make PeriodIndex work like all the others.  To do that, we need to simplify _simple_new, which means fixing the places where inappropriate `values` get passed to it, i.e. #23095.